app/notification/template_types.py

```diff
@@ -2,10 +2,17 @@
 
 
 def email_recipient(data: dict, example_data: dict, notification_class):
+    """
+    Function matches with the correct template type &
+    calls the relevant function from process_notification_data.py.
+    """
     match data.get("type"):
         case "MAGIC_LINK":
             return notification_class.send_magic_link(data)
 
+        case "APPLICATION_RECORD_OF_SUBMISSION":
+            return notification_class.send_application(data)
+
         case "NOTIFICATION" | "REMINDER" | "AWARD":
```

Comment:
💯 Lovely use of cases (pattern matching) :)))

Reply:
😄  Thank you